netty/src/main/java/io/grpc/netty/NettyServerHandler.java

```diff
@@ -526,6 +553,34 @@ public void onRstStreamRead(ChannelHandlerContext ctx, int streamId, long errorC
     }
 
     @Override
+    public void onPingRead(ChannelHandlerContext ctx, ByteBuf data) throws Http2Exception {
+      if (keepAliveManager != null) {
+        keepAliveManager.onDataReceived();
+      }
+      if (!keepAliveEnforcer.pingAcceptable()) {
+        ByteBuf debugData = ByteBufUtil.writeAscii(ctx.alloc(), "too_many_pings");
+        goAway(ctx, connection().remote().lastStreamCreated(), Http2Error.ENHANCE_YOUR_CALM.code(),
+            debugData, ctx.newPromise());
+        flush(ctx);
```

Comment:
No. ctx.flush() goes to the _next_ handler in the pipeline and so wouldn't actually flush anything from HTTP/2. This problem is mostly limited to HTTP/2. Welcome to http2+Netty.